test/service_manager_test.dart

```diff
@@ -0,0 +1,183 @@
+import 'dart:async';
+import 'package:devtools/eval_on_dart_library.dart';
+import 'package:devtools/globals.dart';
+import 'package:devtools/service_manager.dart';
+import 'package:devtools/vm_service_wrapper.dart';
+import 'package:flutter_tools/src/base/file_system.dart';
+import 'package:test/test.dart';
+import 'package:vm_service_lib/vm_service_lib.dart';
+
+import 'test_driver.dart';
+
+void main() {
+  group('serviceManagerTests', () {
+    FlutterRunTestDriver _flutter;
+    VmServiceWrapper service;
+
+    setUp(() async {
+      _flutter =
+          FlutterRunTestDriver(fs.directory('test/fixtures/flutter_app'));
+
+      await _flutter.run(withDebugger: true);
+      service = _flutter.vmService;
+
+      setGlobal(ServiceConnectionManager, new ServiceConnectionManager());
+
+      final completer = new Completer<Null>();
+      await serviceManager.vmServiceOpened(service, completer.future);
```

Comment:
await serviceManager.vmServiceOpened(service, new Completer().future);
I think that makes it clearer that the completer really doesn't matter

Reply:
or comment what completer is and why we intentionally pass in a future that never completes.

Reply:
Done.